app/src/main/java/com/example/travelapp/Itinerary.java

```diff
@@ -16,9 +26,27 @@
     public static final String KEY_LOCATIONS = "locations";
     public static final String KEY_IDS = "ids";
     public static final String KEY_USER = "user";
+    public static final String KEY_TOTAL_DISTANCE = "totalDistance";
+    private static final String TAG = "Itinerary";
+    private static final String KEY_TITLE = "title";
+    private static final String KEY_DETAILS = "details";
+    private Distance distance = new Distance();
 
     public String getLocations() {
-        return getString(KEY_LOCATIONS);
+        JSONArray locationsArray = getJSONArray(KEY_LOCATIONS);
+        String locations = null;
+        for (int i = 0; i < locationsArray.length(); i++) {
+            try {
+                if (i == 0) {
+                    locations = locationsArray.getString(0);
+                } else {
+                    locations = locations + ", " + locationsArray.getString(i);
+                }
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+        return locations;
```

Comment:
why return a string of comma separated locations instead of just an array of strings?